### ISSUE TITLE
How to get to a minimal working example of a valid H5P file?

### ISSUE THREAD

**Issue by tilman-schieber** [h5p/h5p-boilerplate#41]
I want to build a valid H5P File using the example "hello world" boilerplate in the main branch.

I tried the following
1. use npm to build `dist` 
2. `dist` contains only the js&css so we have to create a working H5P module from `dist`. Consulting the H5P docs, I concluded that I have to put `library.json` and `semantics.json` and the `dist` folder in new directory `H5P.HelloWorld` 
3. create a zip file that contains:
     * `content/content.json` (an empty object, although the docs claim the content is optional, there's an error when I don't include it)
     * the `H5P.HelloWorld` folder as described in step 2
     * `h5p.json` [as described in the h5p specification](https://h5p.org/documentation/developers/json-file-definitions). Using H5P.HelloWorld as the main library.
4. rename the zip file to h5p 

Opening the file in Lumi results in 
```
message: "package-validation-failed:library-file-missing,library-file-missing"
```
I have double checked the spelling of all references to the Library Directory. Is there some additional dependency that has to be added to the h5p zip?

the results of my experiments is here (with a zip extension appended so I can upload it here):
[`minimal.h5p.zip`](https://github.com/h5p/h5p-boilerplate/files/11100515/minimal.h5p.zip)

**Comment by otacke** [comment]
@tilman-schieber You'd use the [H5P CLI](https://h5p.org/h5p-cli-guide) tool to **pack** everything. You should not do this manually.

**Comment by tilman-schieber** [comment]
Thank you Oliver, 

this fixes `package-validation-failed:library-file-missing,library-file-missing` 
I was missing the version number at the end of the library subdirectory.

I checked out the boilerplate library again, built it and then packaged it with the H5P tool. (library only)
Then I created another version with h5p.json and the content-directory added and it gave the same result. (library&content)

Both version now give another error when opened in Lumi (in the h5peditor script's `trycatch.ts` in some wrapper fuction around ajax requests)

**Comment by otacke** [comment]
Why would you try to build content manually? The editor will do that for you when you create content. If you do build it manually, make sure that you have a proper `h5p.json` file and a proper `content.json`. And use the -DX flags if you zip content yourself. Answered that plenty of times on the forum: https://h5p.org/comment/43244#comment-43244

H5P doesn't use TypeScript, so the error message that you get is from Lumi.

I suppose it's best if you use the forum for those kinds of questions by the way, not issues in this repository.

**Comment by tilman-schieber** [comment]
I built content manually because I tried to follow the instructions on the official H5P website. And they did exactly that.
As these instructions did not work anymore I tried to find other sources of information about changes to the H5P format that had not been documented. This included this repository.

This appears to me to be a quite sensible train of thought but I seem to be misunderstanding how h5p development ist supposed to work.
So I'll close this and let it be. Thanks for your efforts to help.
